src/modules/json.wren

```diff
@@ -0,0 +1,246 @@
+  error { _error }
+  options { _options }
+  raw { _raw }
+
+  construct new(raw, options) {
+    _result = {}
+    _error = JsonError.empty()
+    _lastEvent = isInit
+    _raw = raw
+    _options = options
+  }
+
+  begin() {
+    stream_begin(_raw)
+    _result = process(next)
+  }
+
+  end() {
+    stream_end()
+  }
+
+  process(event) {
+    _lastEvent = event
+
+    if (event == isError) {
+      _error = JsonError.new(lineno, pos, error_message, true)
+      if (JsonOptions.shouldAbort(_options)) {
+        Fiber.abort("JSON error - line %(lineno) pos %(pos): %(error_message)")
+      }
+      return
+    }
+
+    if (event == isDone) {
+      return
+    }
+
+    if (event == isBoolTrue || event == isBoolFalse) {
+      return (event == isBoolTrue)
+    }
+
+    if (event == isNumeric) {
+      return Num.fromString(this.value)
+    }
+
+    if (event == isString) {
+      return this.value
+    }
+
+    if (event == isNull) {
+      return null
+    }
+
+    if (event == isArray) {
+      var elements = []
+      while (true) {
+        event = next
+        _lastEvent = event
+        if (event == isArrayEnd) {
+          break
+        }
+        elements.add(process(event))
+      }
+      return elements
+    }
+
+    if (event == isObject) {
+      var elements = {}
+      while (true) {
+        event = next
+        _lastEvent = event
+        if (event == isObjectEnd) {
+            break
+        }
+        elements[this.value] = process(next)
+      }
+      return elements
+    }
+  }
+}
+
+class Json {
+
+  // MARK: - File IO
+  // mirroring the FileSystem class
+
+  /**
+  Loads a json file from filesystem.
+  - Signature: load(path: String, options:JsonOptions?) -> <Map|List|Num|String|Null|JsonError>
+  - Parameter path: The file path where the json file is located.
+  - Parameter options: An optional param with options for parsing and encoding json.
+  - Returns: A Wren object or JsonError instance.
+  - Throws: Fiber.abort() if JsonOption.ABORT_ON_ERROR option is passed.
+  */
+  static load(path, options) {
+    var content = FileSystem.load(path)
+    return Json.decode(content, options)
+  }
+
+  static load(path) {
+    return Json.load(path, JsonOptions.ABORT_ON_ERROR)
+  }
+
+  /**
+  Saves a Wren object to a json file in filesystem.
+  - Signature: save(path: String, object:<String|Num|Bool|Map|List|Null>, options:JsonOptions?) -> String
+  - Parameter path: The file path where the json file is located.
+  - Parameter object: The Wren object to convert to a json string.
+  - Parameter options: An optional param with options for parsing and encoding json.
+  - Returns: A json string with the contents of object.
+  */
+  static save(path, object, options) {
+    var content = Json.encode(object, options)
+    FileSystem.save(path, content)
+    return content
+  }
+
+  static save(path, object) {
+    return Json.save(path, object, JsonOptions.ABORT_ON_ERROR)
+  }
+
+  // MARK: - General Manipulation
```

Comment:
What is this line for?

Reply:
In order to make the code easily readable, approachable, and manageable, a best practice to follow is that we divide its different aspects into groups and put a _// MARK:_ above each group with a descriptive comment.

https://nshipster.com/swift-documentation/#mark-todo-fixme

For now is just a simple comment, but maybe in the future would be handy for code analyzing tools.

Reply:
For the moment, this form of notation (as well as other formal documentation comments) aren't in keeping with the existing DOME code style. 

Good code commenting is a best practice across all projects, but it's up to individual projects how that practice is implemented, and to do that here is out of scope for this PR.

I don't mind methods being grouped in related blocks, and indicated as such, I just don't want to commit to a specific tool/system/syntax at this time.

Please remove them for the time being.

Reply:
ok